src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      const animalDiedIndex = Animal.alive.indexOf(this);
+
+      if (animalDiedIndex !== -1) {
+        Animal.alive.splice(animalDiedIndex, 1);
+      }
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
```

Comment:
The condition `animal instanceof Herbivore && animal.hidden === false` assumes that only `Herbivore` instances can be bitten. If there are other subclasses of `Animal` that should also be bitten, this logic might need to be adjusted.